src/promptflow/tests/executor/e2etests/test_batch_engine.py

```diff
@@ -119,6 +120,48 @@ def test_batch_run(self, flow_folder, inputs_mapping, dev_connections):
             assert "line_number" in output, f"line_number is not in {i}th output {output}"
             assert output["line_number"] == i, f"line_number is not correct in {i}th output {output}"
 
+    @pytest.mark.parametrize(
+        "flow_folder, inputs_mapping",
+        [
+            (
+                SAMPLE_FLOW,
+                {"url": "${data.url}"},
+            ),
+            (
+                "prompt_tools",
+                {"text": "${data.text}"},
+            ),
+            (
+                "script_with___file__",
+                {"text": "${data.text}"},
+            ),
+            (
+                "sample_flow_with_functions",
+                {"question": "${data.question}"},
+            ),
+        ],
+    )
+    def test_spawn_mode_batch_run(self, flow_folder, inputs_mapping, dev_connections):
+        os.environ["PF_BATCH_METHOD"] = "spawn"
```

Comment:
For the test case that changes environment variable, you need to create a subprocess to avoid polluting the environment .

Reply:
fixed